lib/mapObjectConstructors/AObjectTypeHandler.cpp

```diff
@@ -95,10 +96,15 @@ void AObjectTypeHandler::init(const JsonNode & input)
 	else
 		aiValue = static_cast<std::optional<si32>>(input["aiValue"].Integer());
 
-	if(input["battleground"].getType() == JsonNode::JsonType::DATA_STRING)
-		battlefield = input["battleground"].String();
-	else
-		battlefield = std::nullopt;
+	battlefield = BattleField::NONE;
+
+	if(!input["battleground"].isNull())
+	{
+		VLC->modh->identifiers.requestIdentifier("battlefield", input["battleground"], [this](int32_t identifier)
+		{
+			battlefield = BattleField(identifier);
```

Comment:
Is it some kind of delayed operation?

Reply:
Yes, all mod identifiers are resolved only after every mod is loaded. This allows mods to be dependent of others.

Reply:
It is not for mod load order, but for object load order.
Delayed callbacks here allow loading of objects even with circular references. 

For example, hero class has reference to faction it belongs to. And faction has reference to chances of each hero class in tavern.
We can't load hero class without loading faction first. But we also can't load faction without loading hero class first.

With these delayed callbacks all objects can be loaded in any order and all references to other objects will be set once loading is complete.